File: backend/modules/shared_libs/src/shared_libs/models/base_models/isoiec_27001.py
from typing import Self

from pydantic import BaseModel, Field, model_validator

from shared_libs.models.base_models.shared.shared.database import PatchBaseModel
from shared_libs.models.model_validators import KbISOIEC27001BaseValidator

__all__ = [
    "ISOIEC27001Model",
    "KbISOIEC27001BaseModel",
]


class ISOIEC27001Model(BaseModel):
    id: str
    annex_ref: str | None = Field(default="")
    category: str | None = Field(default="")
    domain: str | None = Field(default="")
    footnote_ref: str | None = Field(default="")
    guideline_ref: str | None = Field(default="")
    image_name: str | None = Field(default="")
    image_url: str | None = Field(default="")
    policy_ref: str | None = Field(default="")
    in_scope: bool | None = Field(default=False)
    is_auto_compliant: bool | None = Field(default=False)
    std_ref: str | None = Field(default="")
    subdomain: str | None = Field(default="")
    tags: str | None = Field(default="")
    text: str | None = Field(default="")


class KbISOIEC27001BaseModel(
    PatchBaseModel,
    KbISOIEC27001BaseValidator,
):
    isoiec_27001: list["ISOIEC27001Model"] | None = Field(default=[])

    @model_validator(mode="wrap")
    @classmethod
    def validate_model(
        cls,
        data: dict,
        handler,
    ) -> Self:
        model = cls.get_validated_model(
            data=data,
            handler=handler,
        )
        return model

File: frontend/src/interfaces/isoiec27001.ts
import { DatabaseProps } from ".";
import type { MitigationMeasure } from "./register";

//////////////////////////////////////////////////
// NIST CSF Table

//////////////////////////////////////////////////
export interface KbISOIEC27001 extends DatabaseProps {
    isoiec_27001: ISOIEC27001[];
}

export interface ISOIEC27001 {
    id: string;
    //
    annex_ref: string;
    category: string;
    domain: string;
    footnote_ref: string;
    guideline_ref: string;
    image_name: string;
    image_url: string;
    policy_ref: string;
    std_ref: string;
    subdomain: string;
    tags: string;
    text: string;
    // Mitigation measures associated with this policy (backend-provided).
    mitigations?: MitigationMeasure[];
}
